chia/rpc/wallet_rpc_api.py

```diff
@@ -1902,6 +1929,37 @@ async def nft_calculate_royalties(self, request) -> EndpointResult:
             {asset["asset"]: uint64(asset["amount"]) for asset in request.get("fungible_assets", [])},
         )
 
+    async def nft_sign_message(self, request) -> EndpointResult:
+        """
+        Given a NFT ID, sign the message by the NFT's private key.
+        :param request:
+        :return:
+        """
+        if request["nft_id"].startswith("nft"):
+            nft_id: bytes32 = decode_puzzle_hash(request["nft_id"])
+        else:
+            nft_id = bytes32.from_hexstr(request["nft_id"])
+        nft_wallet: Optional[WalletProtocol] = None
+        nft: Optional[NFTCoinInfo] = None
+        for wallet in self.service.wallet_state_manager.wallets.values():
+            if wallet.type() == WalletType.NFT.value:
+                assert isinstance(wallet, NFTWallet)
+                nfts: List[NFTCoinInfo] = await wallet.get_current_nfts()
+                for nft in nfts:
+                    if nft.nft_id == nft_id:
+                        nft_wallet = wallet
+                        nft = nft
+                        break
```

Comment:
`wallet.get_current_nfts()` could be really expensive, right?

Couldn't we use `wallet.get_nft(nft_id)` instead?